### PR TITLE
Write tests and functionality for fewest and most tackles.

### PR DESCRIPTION
**Pull request by DavidHoltkamp1**
Added tackles to initialize method in GameTeams class for access in fewest and most tackles methods. 
Wrote tests for fewest and most tackles, as well as a helper method to determine total tackles for a specific team by season. 

Implemented that functionality (total tackles, most tackles, fewest tackles) in Season class. 
Called those methods as well in our stat tracker class.

### CODE REVIEW

File: lib/game_teams.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ class GameTeams
 
   extend Loadable
 
-  attr_reader :game_id, :team_id, :hoa, :result, :goals, :head_coach, :shots
+  attr_reader :game_id, :team_id, :hoa, :result, :goals, :head_coach, :shots, :tackles
```
Comment (DavidHoltkamp1) [inline lib/game_teams.rb]: added tackles to initialize method here in gameteams to then give me access to that data in season class.

File: lib/season.rb [hunk 2]
```diff
@@ -77,4 +77,24 @@ def least_accurate_team(season_id)
     Team.team_id_to_team_name(least_accurate_team[0])
   end
 
-end
\ No newline at end of file
+  def total_tackles_per_team(season_id)
+    game_per_team_per_season = teams_for_season(season_id).group_by {|game_team| game_team.team_id}
```
Comment (DavidHoltkamp1) [inline lib/season.rb]: Helper method here to reduce (wink wink) redundancy in class and used this in both most and fewest tackles method.

File: test/season_test.rb [hunk 3]
```diff
@@ -44,4 +44,17 @@ def test_it_can_find_the_least_accurate_team
...
 
-end
\ No newline at end of file
+  def test_most_tackles_by_team_by_season
+    assert_equal "FC Dallas", @season.most_tackles("20122013")
+  end
+
+  def test_fewest_tackles_by_team_by_season
+    assert_equal "New York Red Bulls", @season.fewest_tackles("20122013")
+  end
```
Comment (DavidHoltkamp1) [inline test/season_test.rb]: Are these tests robust enough?

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -130,8 +129,15 @@ def best_defense
     @game_collection.best_defense
   end
 
+  def most_tackles(season_id)
+    @season.most_tackles(season_id)
+  end
+
```
Comment (DavidHoltkamp1) [inline lib/stat_tracker.rb]: Serving up 93.67% coverage in stat_tracker.

File: lib/game_teams.rb (hunk 1 above)
Comment (palworth) [reply lib/game_teams.rb]: Necessary step

File: lib/season.rb (hunk 2 above)
Comment (palworth) [reply lib/season.rb]: creating this collection will help with all tackle related algorithms

**Comment by palworth** [comment]
Adding tackle methods to master. All tests passing.

**Pull request merged**